geopandas/io/tests/test_file.py

```diff
@@ -243,6 +243,20 @@ def test_to_file_schema(tmpdir, df_nybb):
     assert result_schema == schema
 
 
+def test_to_file_column_len(tmpdir, df_points):
+    """
+    Ensure that a warning about truncation is given when a geodataframe with
+    column names longer than 10 characters is saved to shapefile
+    """
+    tempfilename = os.path.join(str(tmpdir), "test.shp")
+
+    df = df_points.iloc[:1].copy()
+    df["0123456789A"] = ["the column name is 11 characters"]
+
+    with pytest.warns(UserWarning):
```

Comment:
Would you mind adding `match` to test for this specific warning?

```suggestion
   with pytest.warns(
        UserWarning, match="Column names longer than 10 characters will be truncated"
    )
```